textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -0,0 +1,166 @@
+import edu.uci.ics.textdb.common.field.Span;
+import edu.uci.ics.textdb.common.field.StringField;
+import edu.uci.ics.textdb.common.field.TextField;
+import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.dataflow.common.KeywordPredicate;
+
+/**
+ *  @author prakul
+ *
+ */
+public class KeywordMatcher implements IOperator {
+    private final KeywordPredicate predicate;
+    private ISourceOperator sourceOperator;
+    private ArrayList<Pattern> tokenPatternList;
+    private List<Span> spanList;
+    private String query;
+    private List<Attribute> attributeList;
+    private ArrayList<String> queryTokens;
+    private boolean spanSchemaDefined = false;
+    private Schema spanSchema;
+
+    public KeywordMatcher(IPredicate predicate, ISourceOperator sourceOperator) {
+        this.predicate = (KeywordPredicate)predicate;
+        this.sourceOperator = sourceOperator;
+    }
+
+    @Override
+    public void open() throws DataFlowException {
+        try {
+            sourceOperator.open();
+            query = predicate.getQuery();
+            attributeList = predicate.getAttributeList();
+            queryTokens = predicate.getTokens();
+            tokenPatternList = new ArrayList<Pattern>();
+            Pattern pattern;
+            String regex;
+            for(String token : queryTokens){
+                regex = "\\b" + token.toLowerCase() + "\\b";
+                pattern = Pattern.compile(regex);
+                tokenPatternList.add(pattern);
+            }
+            spanList = new ArrayList<>();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @about Gets next matched tuple. Returns a new span tuple including the
+     *        span results. Performs a scan based search or an index based search depending
+     *        on the sourceOperator provided while initializing KeywordPredicate.
+     *        It scans documents returned by sourceOperator for provided keywords.
+     *
+     * @overview  For each tuple returned by the the sourceOperator loop
+     *           through all the fields in the attributeList. For each field, loop through all the
+     *           matches. Returns only one tuple per document. If there are
+     *           multiple matches, all spans are included in a list. Java Regex
+     *           is used to match word boundaries. Ex : If 'String' name is "lin merry" and
+     *           the description 'Text' is
+     *           "Lin is like Angelina and is a merry person" and the Query is "Lin merry",
+     *           matches should include "lin merry","Lin","merry" but not Angelina.
+     */
+    @Override
+    public ITuple getNextTuple() throws DataFlowException {
+
+        List<IField> fieldList;
+        boolean foundFlag = false;
+        int positionIndex = 0; // Next position in the field to be checked.
```

Comment:
I believe we can move "int positionIndex" and "spanStartPosition" into the `for(int attributeIndex = 0;...` loop.